### ISSUE TITLE
Use tauri v2

### ISSUE THREAD

**Issue by fossdd** [cinnyapp/cinny-desktop#398]
### Describe the problem

tauri v1 (which is currently used by cinny) uses webkit2gtk 4.0. However webkit2gtk 4.0 is outdated and unavailable on many distribitions like Debian or Alpine (which I was packaging Cinny on).

Unfortunaly I have to disable the cinny package on Alpine Linux since we removed webkit2gtk 4.0 and only support webkit2gtk 4.1 or 6.0.

See also https://github.com/cinnyapp/cinny-desktop/issues/367 or https://github.com/tauri-apps/tauri/issues/9662

### Describe the solution you'd like

Ideally cinny-desktop could be moved to tauri v2 which supports webkit2gtk 4.1 and would restore support for many distributions.

**Comment by Neved4** [comment]
I like this idea!

Could be done by using the commands from [Upgrade from Tauri 1.0 | Automated Migration](https://v2.tauri.app/start/migrate/from-tauri-1/#automated-migration).

We'll need to remember to upgrade our plugins under `Cargo.toml`.

**Comment by Neved4** [comment]
For reference this is currently blocking https://github.com/cinnyapp/cinny-desktop/issues/247, https://github.com/cinnyapp/cinny/issues/446

I'll take care of those when we upgrade [cinny-desktop](https://github.com/cinnyapp/cinny-desktop/)

**Comment by Neved4** [comment]
Gave it a try, and attempted to migrate tauri `v1` to `v2`.

Pulled off the right dependencies, and [tauri.conf.json](https://github.com/cinnyapp/cinny-desktop/blob/main/src-tauri/tauri.conf.json) had to undergo some modifications for `v2`. Eventually hit a block, since the migration produces numerous Rust errors. Couldn't seem to solve all of them. Clearly this is going to take much longer for me to address.

Overall learned some things, but can't do it yet on my own. Could set an initial PR, but would be broken atm.

Any help would be awesome.

**Comment by fossdd** [comment]
Thanks for the effort!

**Comment by kfiven** [comment]
@Neved4 I also had similar rust issues. Same, thought this would be easy task but turned out to be much complicated. For a reference you could find some other other projects that migrated to v2.

**Comment by bengaywins** [comment]
My first rust project contribution, but the above linked PR was able to build on both WSL and Arch. I am just a lowly ops dude, so maybe the _small_ changes to the `main.rs` aren't correct (for some reason I don't understand), but it at least worked when I was using the updated client under testing last night.

**Comment by ogarcia** [comment]
The migration to Tauri 2 is now even more necessary since [libsoup 2 is deprecated](https://webkitgtk.org/2025/10/07/webkitgtk-soup2-deprecation.html) and all related packages will be removed from distributions. This will cause cinny-desktop to fail to start.

**Comment by ryand56** [comment]
webkitgtk_4_0 is now in the process of being dropped from nixpkgs. Is there any immediate progress on this? cinny-desktop will have to be removed from nixpkgs if necessary.

**Comment by Neved4** [comment]
@ogarcia @ryand56 See https://github.com/cinnyapp/cinny-desktop/pull/429

**Comment by ogarcia** [comment]
@Neved4 thanks for the note, but both ryand56 and I have already commented on that PR and are already aware of it.

The important thing is not that this PR exists (which, incidentally, is obsolete and cannot currently be merged), but that it is actually merged and a version with Tauri V2 is released.

**Comment by jficz** [comment]
> a version with Tauri V2 is released.

which version would that be? I can't see any mention of Tauri v2 in the change logs but maybe I'm just looking at the wrong place..

**Comment by ogarcia** [comment]
@jficz reread my comment and don't take it out of context. There is still no version with Tauri 2 (which is what is requested in this issue). The only thing available is an obsolete MR (#429) that is currently no longer valid.

**Comment by jficz** [comment]
@ogarcia sorry, your comment's tense can be read both as "it needs to be done" and "it has been done", no flame intended, just clarifying if I understood correctly (apparently didn't :) )

**Comment by ogarcia** [comment]
@jficz no worries, English isn't my native language, so I might make a mistake here and there 😉 (and I might even sound harsher than I should)

**Comment by q234rty** [comment]
Note that webkitgtk upstream [is removing support for webkit2gtk-4.0](https://webkitgtk.org/2025/10/07/webkitgtk-soup2-deprecation.html) in the future 2.52.0 release.

**Comment by CEbbinghaus** [comment]
> The migration to Tauri 2 is now even more necessary since [libsoup 2 is deprecated](https://webkitgtk.org/2025/10/07/we.bkitgtk-soup2-deprecation.html) and all related packages will be removed from distributions. This will cause cinny-desktop to fail to start.

Absolutely already true, this is preventing the use of cinny in the latest versions of nixos

**Comment by Kruemmelspalter** [comment]
what is the current state of this? what is missing in the current PRs?

**Comment by PukPukov** [comment]
> Absolutely already true, this is preventing the use of cinny in the latest versions of nixos

On nixos you can just pick cinny from warbler. You would need to compile it though

**Comment by jficz** [comment]
> On nixos you can just pick cinny from warbler.

yeah, but you'd have to compile half of the world, including many unsupported and _insecure_ libs. Not sure that's the best idea but ymmv

**Comment by jtrees** [comment]
This seems to have been implemented in #496.
